app/controllers/arrangement/ArrangementCheckYourAnswersController.scala

```diff
@@ -52,9 +52,9 @@ class ArrangementCheckYourAnswersController @Inject()(
 
       val list: Seq[SummaryList.Row] =
         Seq(helper.whatIsThisArrangementCalledPage(id)
-          , helper.whatIsTheImplementationDatePage(id)).flatten ++
-          helper.buildReportingThisArrangement(id) ++
-          Seq(helper.whichExpectedInvolvedCountriesArrangement(id)
+          , helper.whatIsTheImplementationDatePage(id)
+          , helper.buildWhyAreYouReportingThisArrangementNow(id)
+          , helper.whichExpectedInvolvedCountriesArrangement(id)
```

Comment:
have you verified that it's not showing empty rows? which I seem to remember was the point of the flattening

Reply:
The flatten combinator was actually moved to the final, line 60. Previously I had one in the middle because a method was returning an intermediary Seq , which is no longer necessary.